src/Microsoft.VisualStudio.Threading/AsyncReaderWriterLock.cs

```diff
@@ -43,9 +43,9 @@ public partial class AsyncReaderWriterLock : IDisposable
         private readonly object syncObject = new object();
 
         /// <summary>
-        /// The synchronization context applied to folks who hold upgradeable read and write locks.
+        /// The default SynchronizationContext to schedule work after issuing a lock.
         /// </summary>
-        private readonly NonConcurrentSynchronizationContext nonConcurrentSyncContext = new NonConcurrentSynchronizationContext();
+        private readonly SynchronizationContext defaultSynchronizationContext = new SynchronizationContext();
```

Comment:
The default `SynchronizationContext` simply schedules work on the thread pool. Can we just remove this field altogether and use `ThreadPool.QueueUserWorkItem` wherever you had the code using this new default one?

Reply:
Looking further down it appears that you're basically using this as a `non-null` value so you know now to initialize another field repeatedly. That may be a legit use for it. But can you perhaps add a comment?
Also, I think you can add `static` to this field so we only allocate one of these regardless of how many instances of this class we create.

Reply:
Yes, just to prevent another field to mark that state.  I added comments.